tests/test_unit.py

```diff
@@ -1984,5 +1985,18 @@ def test_custom_fields_parser(self):
             assert is_deep_object is True
 
 
+class CustomRegexMethodTestCase(TestCase):
+    def test_custom_regex_escape(self):
+        for regex_input, output in (
+            (".?well+", "\\.\\?well\\+"),
+            ("&[^abc]~", "\\&\\[^abc\\]\\~"),
+            ("some|*#thing", "some\\|\\*\\#thing"),
+            ('.?+*|{}[]()"\\#@&<>~', '\\.\\?\\+\\*\\|\\{\\}\\[\\]\\(\\)\\"\\\\\\#\\@\\&\\<\\>\\~'),
+            ("\\", "\\\\"),
+        ):
+            result = custom_escape(regex_input)
```

Comment:
This works fine as is, but i think the cause of the test failure is easier to track down when you wrap each assert in a subtest. It gives the the test case it failed on in the header which I find to be helpful.
```suggestion
        with self.subTest(regex_input=regex_input):
            result = custom_escape(regex_input)
```

Reply:
done! thanks! :+1: